### PR TITLE
Disable stats recorder for built-in PulsarClient

### PR DESCRIPTION
**Pull request by BewareMyPower**
### Motivation

In `PulsarService`, there's a built-in client used in many places, like topic compaction, system topics read/write. It could also be used in protocol handlers. Different with the manually created Pulsar client, it can read the configured authentication params and TLS related configs for broker-to-broker communication.

However, it doesn't change the default `statsIntervalSeconds` config, which means each time a producer/consumer/reader is created, a stats recorder (`ProducerStatsRecorderImpl` or `ConsumerStatsRecorderImpl`) will be created, in which there's a Netty `TimerTask` that prints stats periodically (the default interval is 1 minute). If there're many producers/consumers/readers created by this built-in client, the unnecessary CPU usage will be high.

### Modifications

Configure the `statsIntervalSeconds` with zero value to disable stats recorder.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 1]
```diff
@@ -1327,6 +1327,8 @@ public synchronized PulsarClient getClient() throws PulsarServerException {
                             this.getConfiguration().getBrokerClientAuthenticationPlugin(),
                             this.getConfiguration().getBrokerClientAuthenticationParameters()));
                 }
+
+                conf.setStatsIntervalSeconds(0);
```
Comment (hangc0276) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:1331]: For the pulsar replicator, it also call this `getClient` method to init pulsar client, which should retain the stats. So we'd better add a parameter to decide whether keep the client stats.

Comment (BewareMyPower) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:1331]: Why do we need the stats in replicator?

**Review by michaeljmarshall — APPROVED** [review]
LGTM.

If we find that we are interested in these stats, I think it makes more sense to update the implementation to add them to the results returned by the metrics endpoint instead of hide the stats in logs. Technically, this could be useful for client applications as well. The pulsar go client creates prometheus metrics, and I've found those very useful.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java (hunk 1 above)
Comment (hangc0276) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:1331]: Ok, it doesn't need the stats in replicator

**Review by hangc0276 — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Review by 315157973 — APPROVED** [review]

**Comment by 315157973** [comment]
My only concern is that some users conduct some business analysis by collecting this log. At least this change needs to be explained in the Release, right?

**Comment by BewareMyPower** [comment]
> My only concern is that some users conduct some business analysis by collecting this log. At least this change needs to be explained in the Release, right?

Yeah, you're right. Anyway it's a trade-off. And I'm curious about how can we explain it in release notes?

**Pull request merged**